### PR TITLE
[VIVO-1671] Restrict context node labels added to search index to relationships

### PR DESCRIPTION
**Pull request by gneissone**
**[JIRA Issue](https://jira.duraspace.org/browse/VIVO-1671)**: https://jira.duraspace.org/browse/VIVO-1671

# What does this pull request do?
Restricts addition of labels to the search index for context nodes (aka faux properties).

# What's new?
Labels will be added to a search document for objects connected via a faux property if the property is one of the following:

 Advising Relationship (vivo)
 Authorship (vivo)
 Awarded Degree (vivo)
 Award or Honor Receipt (vivo)
 Contract (vivo)
 Editorship (vivo)
 Grant (vivo)
 Issued Credential (vivo)
 Position (vivo)

which roll up to vivo:Relationship.

e.g. take this relationship:
person vivo:relatedBy authorship
authorship a vivo:Authorship
authorship vivo:relates publication

In this case, ALLTEXT in the search document for person will include the label for the publication.
vs. this relationship:
publication1 vivo:relatedBy organization
instrument a vivo:Instrument
instrument vivo:relates (many, many publications)

the label of the many, many publications will NOT be added to the search document for original publication because the class type of is vivo:Instrument, not one of the relationships listed above.

# How should this be tested?
Before the changes, add triples like above using vivo:relates or vivo:relatedBy that does not connect two objects via a vivo:Relationship. The label for the many, many publications (and the instrument) will be added to the search document for the original publication1. Apply the changes, and now only the label for instrument will be added. 

# Additional Notes:
In summary, VIVO tries to be smart about adding labels for faux properties, but without this restriction it gets overzealous if a site is using vivo:relates/vivo:relatedBy for anything besides faux properties.

# Interested parties
@Stefan-Wolff

### CODE REVIEW

**Comment by gneissone** [comment]
@awoods, done.

**Comment by awoods** [comment]
@gneissone : regarding tooling for making whitespace changes invisible, I agree, they exist. To the degree that we can, I prefer to keep commits cleaner by not introducing whitespace changes in the first place.
If it becomes too much of a trouble to keep whitespace updates out of commits, I am happy to revisit.

Thanks for your follow-on commit!

**Comment by awoods** [comment]
@Stefan-Wolff : If you approve of this pull-request, we can get it merged.
